lib/active_model/serializer.rb

```diff
@@ -69,56 +43,98 @@ def self.digest_caller_file(caller_line)
       serializer.class_attribute :_cache_digest # @api private : Generated
     end
 
-    # Serializers inherit _attributes and _attributes_keys.
+    # Serializers inherit serialized_attributes, _attributes_keys, and _reflections.
     # Generates a unique digest for each serializer at load.
     def self.inherited(base)
       caller_line = caller.first
-      base._attributes = _attributes.dup
+      base.serialized_attributes = serialized_attributes.dup
       base._attributes_keys = _attributes_keys.dup
       base._links = _links.dup
       base._cache_digest = digest_caller_file(caller_line)
       super
     end
 
-    # @example
-    #   class AdminAuthorSerializer < ActiveModel::Serializer
-    #     type 'authors'
-    def self.type(type)
-      self._type = type
+    # Matches
```

Comment:
As we're already moving things around in this PR, shouldn't we use the opportunity to extract all caching-related methods/constants into its own module? It would make this file less confusing IMHO.